libhoney/transmission.py

```diff
@@ -84,19 +84,33 @@ def _send(self, ev):
             "X-Honeycomb-Team": ev.writekey,
             "X-Honeycomb-SampleRate": str(ev.sample_rate)})
         preq = self.session.prepare_request(req)
-        resp = self.session.send(preq)
-        if (resp.status_code == 200):
-            sd.incr("messages_sent")
-        else:
+        try:
+            resp = self.session.send(preq)
+            if (resp.status_code == 200):
+                sd.incr("messages_sent")
+            else:
+                sd.incr("send_errors")
+            dur = get_now() - start
+            response = {
+                "status_code": resp.status_code,
+                "duration": dur.total_seconds() * 1000,  # report in milliseconds
+                "metadata": ev.metadata,
+                "body": resp.text,
+                "error": "",
+            }
+        except requests.exceptions.ConnectionError as e:
+            # sometimes the ELB returns SSL issues for no good reason.
+            # catch these and push back a broken response so the event can be
+            # properly handled by the calling code
+            dur = get_now() - start
             sd.incr("send_errors")
-        dur = get_now() - start
-        response = {
-            "status_code": resp.status_code,
-            "duration": dur.total_seconds() * 1000,  # report in milliseconds
-            "metadata": ev.metadata,
-            "body": resp.text,
-            "error": "",
-        }
+            response = {
+                "status_code": 0,
+                "duration": dur.total_seconds() * 1000,  # report in milliseconds
+                "metadata": ev.metadata,
+                "body": "",
+                "error": repr(e),
+            }
```

Comment:
This duplication makes me a little sad - would it be feasible to move calculating the duration and building the response object to a `finally` clause? (The exception `e` should still be in scope in the `finally` clause, so you could do `"error": repr(e) if e else ""`.)